nix/pkgs/xcodeenv/compose-xcodewrapper.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, lib, writeShellScriptBin }:
+{ version ? "15.1"
+, allowHigher ? false
+, xcodeBaseDir ? "/Applications/Xcode.app" }:
+
+assert stdenv.isDarwin;
+
+let
+  xcodebuildWrapper = writeShellScriptBin "xcodebuild" ''
+    currVer=$("${xcodeBaseDir}/Contents/Developer/usr/bin/xcodebuild" -version | head -n1)
+    ${if allowHigher then ''
+    if [ -z "$(printf '%s\n' "${version}" "$currVer" | sort -V | head -n1)""" != "${version}" ]
```

Comment:
either its cache or this `allowHigher ` logic doesn't work as expected. I'm refactoring this if else clause now